### PR TITLE
feat(content-api-kit): add reusable Express app kit for content APIs

### PR DESCRIPTION
**Pull request by ericsen-tsai**
## Summary

Introduces `@kids-reporter/content-api-kit`, a publishable workspace package that centralizes Express app scaffolding (CORS modes, JSON parsing, logging, error helpers) and Go API JWT auth. Refactors `content-api` to consume the kit, and broadens CI/root test orchestration so every workspace with a real `test` script runs in CI.

## Changes

- Add `packages/content-api-kit` with Babel + TypeScript build, README, and exports for `createContentApiApp`, `createLoggerMw`, `asyncRoute`, `sendJsonError`, `defaultErrorHandler`, and `verifyGoApiJwt` / `verifyGoApiAccessToken`
- Move request logger middleware from `content-api` into the kit; support `CreateLoggerMwOptions` (`projectId`, `slowThresholdMs`)
- Extract Go API JWT verification into the kit with configurable `secret` / `issuer` / `audience` and optional `onReject` callback (no built-in logger coupling)
- Refactor `content-api` `createApp()` to use `createContentApiApp()` with route-level CORS (`public` vs `credentialed` for `/auth`)
- Wire v1 member routes to `verifyGoApiJwt({ ..., onReject })` with `emitStructured` logging preserved at the app layer
- Remove duplicated `go-api-jwt.ts`, `verify-go-api-jwt.ts`, middleware index, and in-package JWT tests from `content-api`
- Add `scripts/run-workspace-tests.mjs` and root `yarn test` to run Vitest (and other test scripts) across all workspaces that define a non-placeholder `test` script
- Rename CI job `content-api-test` → `test`; run `yarn test` instead of only `@kids-reporter/content-api` tests
- Update `content-api` Makefile `build-deps` to build `content-api-kit`
- Extend root ESLint `files` glob to include `scripts/**/*.{js,mjs}`

## Additional Notes

- JWT middleware is now factory-based (`verifyGoApiJwt(opts)`); consumers must pass env-backed options and wire `onReject` fortructured auth-failure logs (behavior matches prior `content-api` logging when configured as in v1 routes).
- Kit JWT unit tests are slimmer than the removed `content-api` suite (fewer mocked error-classification cases); edge-case coverage relies more on integration tests in `content-api`.
- CI now runs all workspace test scripts sequentially; adding slow or flaky test workspaces will lengthen the `test` job.
- Ensure `content-api-kit` is built before deploying `content-api` (`make build-deps` / CI build order).
- **Risk:** Any downstream service copying old `verifyGoApiJwt(req, res, next)` signature must migrate to the factory + options pattern.

## Reference(s)

### CODE REVIEW

File: packages/content-api/src/app.ts [hunk 1]
```diff
@@ -17,6 +18,14 @@ import { sendJsonError } from './utils/send-json-error.js'
 const errors = _errors.default
 const statusCodes = consts.statusCodes
 
+function joinBasePath(basePath: string, path: string): string {
+  const trimmedBase = basePath.replace(/\/+$/, '')
+  const normalizedBase = trimmedBase === '/' ? '' : trimmedBase
+  if (!normalizedBase) return path
+  if (!path || path === '/') return normalizedBase
+  return `${normalizedBase}${path.startsWith('/') ? '' : '/'}${path}`
+}
```
[the remark this answers is not in the document]
Comment (ericsen-tsai) [reply packages/content-api/src/app.ts]: Fixed

File: packages/content-api-kit/src/auth/go-api-jwt.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  return 'jwt_verification_failed'
+}
+
+export function verifyGoApiAccessToken(
+  token: string,
+  opts: GoApiJwtOptions
+): GoApiAccessTokenPayload {
+  if (!opts.secret) {
+    throw new Error('GO_API_JWT_SECRET is not configured')
+  }
```
[the remark this answers is not in the document]
Comment (ericsen-tsai) [reply packages/content-api-kit/src/auth/go-api-jwt.ts]: Fixed

File: packages/content-api-kit/src/auth/go-api-jwt.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    const reject = (
+      reason: GoApiJwtAuthFailureReason,
+      extra?: { jwtLibraryErrorName?: string }
+    ) => {
+      opts.onReject?.({
+        reason,
+        path: req.originalUrl,
+        method: req.method,
+        ...extra,
+      })
```
[the remark this answers is not in the document]
Comment (ericsen-tsai) [reply packages/content-api-kit/src/auth/go-api-jwt.ts]: Fixed

File: packages/content-api-kit/src/middlewares/logger.ts [hunk 4]
```diff
@@ -25,12 +29,26 @@ function getGlobalLogFields(req: express.Request, projectId: string) {
...
-export function createLoggerMw(projectId: string): express.RequestHandler {
-  const handler: express.RequestHandler = (req, res, next) => {
-    const { globalLogFields, traceContext } = getGlobalLogFields(req, projectId)
+export function createLoggerMw(
+  projectIdOrOpts: string | CreateLoggerMwOptions
+): express.RequestHandler {
+  const opts: CreateLoggerMwOptions =
+    typeof projectIdOrOpts === 'string'
+      ? { projectId: projectIdOrOpts }
+      : projectIdOrOpts
```
[the remark this answers is not in the document]
Comment (ericsen-tsai) [reply packages/content-api-kit/src/middlewares/logger.ts]: Fixed

File: packages/content-api-kit/src/utils/send-json-error.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  message: string,
+  details?: unknown
+) {
+  res.status(status).json({
+    error: {
+      code,
+      message,
+      ...(details !== undefined ? { details } : {}),
+    },
+  } satisfies JsonErrorBody)
```
[the remark this answers is not in the document]
Comment (ericsen-tsai) [reply packages/content-api-kit/src/utils/send-json-error.ts]: fixed

File: packages/content-api/src/app.ts [hunk 6]
```diff
@@ -127,7 +95,31 @@ export function createApp({
...
 
-  app.use(errorHandler)
-
-  return app
+  return createContentApiApp({
+    basePath,
+    corsAllowOrigin,
+    jsonLimit: '1mb',
+    loggerMiddleware: createLoggerMw({
+      projectId: gcpProjectId,
```
[the remark this answers is not in the document]
Comment (ericsen-tsai) [reply packages/content-api/src/app.ts:103]: fixed, since currently the credential set per route, I removed this option entirely.

File: scripts/run-workspace-tests.mjs [hunk 7]
```diff
@@ -0,0 +1,76 @@
+/* eslint-env node */
+import fs from 'node:fs/promises'
+import path from 'node:path'
+import { fileURLToPath } from 'node:url'
+import { execFileSync } from 'node:child_process'
```
[the remark this answers is not in the document]
Comment (ericsen-tsai) [reply scripts/run-workspace-tests.mjs]: Fixed

File: packages/content-api/src/routes/v1/v1-members.ts [hunk 8]
```diff
@@ -50,7 +51,23 @@ const requireUserId = (
...
+          severity: 'WARNING',
+          message: 'Go API JWT request rejected',
+          goApiJwtAuthFailureReason: info.reason,
+          path: info.path,
+          method: info.method,
+          jwtLibraryErrorName: info.jwtLibraryErrorName,
+        })
+      },
+    })
+  )
```
[the remark this answers is not in the document]
Comment (ericsen-tsai) [reply packages/content-api/src/routes/v1/v1-members.ts:70]: Fixed

File: packages/content-api/src/routes/v1/v1-qna-members.ts [hunk 9]
```diff
@@ -104,7 +106,23 @@ function parsePathIdParam(
...
+          severity: 'WARNING',
+          message: 'Go API JWT request rejected',
+          goApiJwtAuthFailureReason: info.reason,
+          path: info.path,
+          method: info.method,
+          jwtLibraryErrorName: info.jwtLibraryErrorName,
+        })
+      },
+    })
+  )
```
[the remark this answers is not in the document]
Comment (ericsen-tsai) [reply packages/content-api/src/routes/v1/v1-qna-members.ts:125]: Fixed

File: packages/content-api-kit/src/auth/go-api-jwt.ts [hunk 10]
```diff
@@ -0,0 +1,131 @@
+import type { NextFunction, Request, Response } from 'express'
+import jwt from 'jsonwebtoken'
```
[the remark this answers is not in the document]
Comment (ericsen-tsai) [reply packages/content-api-kit/src/auth/go-api-jwt.ts:4]: Fixed
